### PR TITLE
JENKINS-48155: gerrit-trigger-plugin does not repsect dynamicTriggerConfiguration flag

### PR DESCRIPTION
**Pull request by Jimilian**
If user does not want to use dynamic configuration anymore, let's assign
the list of projects to empty list to avoid possible race conditions.

Also instead of firing update with empty url, better to check it.

### CODE REVIEW

File: src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritDynamicUrlProcessor.java [hunk 1]
```diff
@@ -246,13 +245,6 @@ private static String regexEscapted(char symbol) {
     public static List<GerritProject> fetch(String gerritTriggerConfigUrl, String serverName)
             throws IOException, ParseException {
 
-        if (gerritTriggerConfigUrl == null) {
```
Comment (rsandell) [inline src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritDynamicUrlProcessor.java]: I see no reason to remove this, it will just cause another `MalformedURLException` to be thrown from the URL constructor with a more cryptic message.

File: src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java [hunk 2]
```diff
@@ -1731,14 +1733,6 @@ public void updateTriggerConfigURL() {
             String triggerInformationMessage = MessageFormat.format(
                     "ParseException when fetching dynamic trigger url: {0}", pe.getMessage());
             triggerInformationAction.setErrorMessage(triggerInformationMessage);
-        } catch (MalformedURLException mue) {
-            String logErrorMessage = MessageFormat.format(
-                    "MalformedURLException for project: {0} and URL: {1} Message: {2}",
```
Comment (rsandell) [inline src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java]: Why is this removed? The user could still specify a malformed url in the config screen, and now the error will be harder to find.

File: src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java [hunk 3]
```diff
@@ -1720,8 +1723,7 @@ public void updateTriggerConfigURL() {
                 triggerInformationAction.setErrorMessage("Dynamic trigger configuration needs "
                         + "a specific configured server");
             } else {
-                List<GerritProject> fetchedProjects = GerritDynamicUrlProcessor.fetch(triggerConfigURL, serverName);
-                dynamicGerritProjects = fetchedProjects;
+                dynamicGerritProjects = GerritDynamicUrlProcessor.fetch(triggerConfigURL, serverName);
```
Comment (rsandell) [inline src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java]: Why this change?
If there is an error now you will be stuck with not triggering on anything instead of at least triggering on what was parsed in the last fetch.

File: src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTriggerTimerTask.java [hunk 4]
```diff
@@ -59,6 +59,9 @@ public void run() {
         if (trigger == null) {
             return;
         }
+        if (trigger.getTriggerConfigURL() == null || trigger.getTriggerConfigURL().isEmpty()) {
```
Comment (rsandell) [inline src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTriggerTimerTask.java]: use `StringUtils.isEmpty()`

File: src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritDynamicUrlProcessor.java (hunk 1 above)
Comment (Jimilian) [reply src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritDynamicUrlProcessor.java]: Because updateTriggerConfigURL was trigged only from once place and now we have a check directly in this place.
Should I keep it to double check?

File: src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java (hunk 3 above)
Comment (Jimilian) [reply src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java]: if `fetch` will raise a exception - assignment will not happen. Do not see any difference - just use one variable less.

File: src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java (hunk 2 above)
Comment (Jimilian) [reply src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java]: Agree, will restore.

**Comment by Jimilian** [comment]
@rsandell Now it's the most conservative PR ever ;)

File: src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritDynamicUrlProcessor.java (hunk 1 above)
Comment (rsandell) [reply src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritDynamicUrlProcessor.java]: Well it's a public method so it's best to keep it I think :)
